src/functions-and-arrays.js

```diff
@@ -1,18 +1,84 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(x,y) {
+  if (x>=y) return x;
+  if (x<y) return y;
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+function findLongestWord(wordArr) {
+  let l=0;
+  let longestWordIndex=0;
+  for (let i=0; i<wordArr.length; i++) {
+    if (wordArr[i].length>l) {
+      l=wordArr[i].length;
+      longestWordIndex=i;
+    }
+  }
+  return wordArr[longestWordIndex];
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumNumbers(numArr) {
+  let sum=0;
+for (num of numArr) {
+  sum += num;
+  //console.log(sum);
+}
+  return sum;
+}
+
+// Iteration #3.1
+function sum(arr) {
+  let replaceArr = [];
+  for (element of arr) {
+    console.log(typeof element);
+    if (typeof element == "number") {
+      replaceArr.push(element);
+    }
+    if (typeof element == "string") {
+      replaceArr.push(element.length);
+    }
+    if (element === false) {
+      replaceArr.push(0);
+    }
+    if (element === true) {
+      replaceArr.push(1);
+    }
+  }
+  console.log(replaceArr);
+  return sumNumbers(replaceArr);
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+function averageNumbers(avgArr) {
+  let sum=sumNumbers(avgArr);
+  let avg=sum/avgArr.length;
+  return avg;
+}
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+function averageWordLength(lenAvgArr) {
+  let lengthsArr=[];
+  for (word of lenAvgArr) {
+    lengthsArr.push(word.length);
+  }
+  let sum=sumNumbers(lengthsArr);
+  console.log(`${sum} letters in total, over ${lengthsArr.length} words.`);
+  let avg=sum/lengthsArr.length;
+  return avg;
+}
+
+//bonus: mixed average
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+function avg(arr) {
+  return mixedAvg=sum(arr)/arr.length;
```

Comment:
this is the response from Jasmine
`Expected 5.111111111111111 to be 5.11.`
so for that, you can use the method .toFix() like this
```return Number((sum(array) / array.length).toFixed(2));```